Vagrantfile

```diff
@@ -1,93 +1,166 @@
+# Configuration parameters
+VAGRANT_BASE_OS = "bento/ubuntu-24.04" # "bento/ubuntu-22.04"
+PRIVATE_NETWORK = "private_network"    # For Host -> VM and VM <-> VM (within the network)
+BASE_CIDR       = "10.201.0.0"         # Base address for pods
+
+# Create list of one or more Control Plane Nodes (but one is sufficient)
+CPLANE_NODES = [
+  { name: "cplane",  box: VAGRANT_BASE_OS, network: PRIVATE_NETWORK, ip: "192.168.63.11" }
+]
+
+# Create list of one or more worker nodes
+# Mindful of the 'name' and 'ip' values for each
+WORKER_NODES = [
+  { name: "worker1", box: VAGRANT_BASE_OS, network: PRIVATE_NETWORK, ip: "192.168.63.12" }
+]
+
+# Work out the "/etc/hosts" values to get copied in each node (cplanes and workers)
+ALL_NODES = CPLANE_NODES + WORKER_NODES
+ETC_HOSTS = ALL_NODES.map { |n| "#{n[:ip]} #{n[:name]}" }.join("\n") + "\n"
+
 Vagrant.configure("2") do |config|
-  # Define master VM
-  config.vm.define "master" do |master|
-    master.vm.box = "bento/ubuntu-22.04"
-    master.vm.network "private_network", ip: "192.168.63.1"
-    master.vm.hostname = "master"
-    master.vm.provider "virtualbox" do |v|
-      v.name = "master"
-      v.memory = 2048
-      v.cpus = 2
-    end
-    master.vm.provision "shell", inline: <<-SHELL
-      sudo echo "192.168.63.1 master\n192.168.63.2 worker" >> /etc/hosts
-      sudo apt update
-      sudo apt install ca-certificates curl
-      sudo install -m 0755 -d /etc/apt/keyrings
-      sudo curl -fsSL https://download.docker.com/linux/ubuntu/gpg -o /etc/apt/keyrings/docker.asc
-      sudo chmod a+r /etc/apt/keyrings/docker.asc
+  # Define Control Plane Nodes
+  CPLANE_NODES.each do |node|
+    config.vm.define node[:name] do |cplane|
+      cplane.vm.box = node[:box]
+      cplane.vm.network node[:network], ip: node[:ip]
+      cplane.vm.hostname = node[:name]
+      cplane.vm.provider "virtualbox" do |v|
+        v.name = node[:name]
+        v.memory = 2048
+        v.cpus = 2
+      end
+      cplane.vm.provision "shell",
+        env: {
+          "ETC_HOSTS"     => ETC_HOSTS,
+          "BASE_CIDR"     => BASE_CIDR,
+          "API_SERVER_IP" => node[:ip] # API Server is the control plane host itself
+        },
+        inline: <<-SHELL
+        # Add Nodes to /etc/hosts
```

Comment:
In this case it was necessary to use "read" to split off the '\n' in the hosts list and then add each host line on it's own. Otherwise the newline character got messed up